### PR TITLE
Feature/avx 64 add https support

### PR DESCRIPTION
**Pull request by dat-tran-matternet**

### CODE REVIEW

File: ntripclient.c [hunk 1]
```diff
@@ -566,6 +583,104 @@ static int encode(char *buf, int size, const char *user, const char *pwd)
...
+    bool hdr_done = false;
+
+    memset((void*)hdr, 0, sizeof(HttpHeader));
+
+    while (hdr_done == false)
+    {
+        // Read more data into the buffer; append to existing data if any.
+        ssize_t bytes_read = curl_recv_(curl, buf + total_bytes, MAXDATASIZE - total_bytes - 1, 0);
+        if (bytes_read <= 0)
+            return -1;
```
Comment (adriennet823) [inline ntripclient.c]: Might I suggest we use an enum for return codes like these to improve readability?
e.g. 
`typedef enum {`
 `    INVALID_NUM_BYTES_READ = -1,`
 `    VALID_NUM_BYTES_READ = 0,`
`} ReadBufferRetCode;`

and then in main(): 
`ReadBufferRetCode retcode = parse_http_header(curl, &http_hdr);`
`...`
`if (retcode != VALID_NUM_BYTES_READ || ...)`

File: ntripclient.c [hunk 2]
```diff
@@ -566,6 +583,104 @@ static int encode(char *buf, int size, const char *user, const char *pwd)
   return bytes;
 }
 
+static ssize_t curl_send_(CURL *curl, const void *buf, size_t len, int flags)
+{
+    (void)flags; // not used
+    size_t bytes_sent = 0;
+    CURLcode result = curl_easy_send(curl, buf, len, &bytes_sent);
+    return (result == CURLE_OK) ? (ssize_t)bytes_sent : -1;
```
Comment (adriennet823) [inline ntripclient.c:591]: You could also use the suggested `ReadBufferRetCode` enum here as well for invalid bytes sent

Comment (dat-tran-matternet) [reply ntripclient.c:591]: This one I will want to keep the c convention for fucntions like read/write/recv/send which returns -1 on error.

File: ntripclient.c (hunk 1 above)
Comment (dat-tran-matternet) [reply ntripclient.c]: How about I make the return a boolean since it's a all or nothing we really care about?

Comment (adriennet823) [reply ntripclient.c]: Sure that works too 👍

**Review by AydenChubbic — APPROVED** [review]

File: ntripclient.c (hunk 1 above)
Comment (dat-tran-matternet) [reply ntripclient.c]: Done

**Pull request merged**